### PR TITLE
Porting withholding tax 8

### PR DESCRIPTION
**Pull request by eLBati**

### CODE REVIEW

File: l10n_it_withholding_tax/account.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+
+
+class res_company(orm.Model):
```
Comment (veloutin) [inline l10n_it_withholding_tax/account.py]: Should this belong in company.py?

**Comment by veloutin** [comment]
There is a mix of under_score and CamelCase in class names. Also, it would be a good time to switch some of the methods to the v8 api if possible.

**Comment by eLBati** [comment]
Thanks @veloutin I improved classes and files names

**Comment by nicothebrush** [comment]
Need account_voucher_cash_basis as dependency, currently unported in OCA/account-payment, here:
https://github.com/OCA/account-payment/tree/8.0/__unported__
WIP here:
https://github.com/tafaRU/account-payment/tree/8.0-account_vat_on_payment-migr

**Comment by eLBati** [comment]
@nicothebrush done, thanks

**Comment by lepistone** [comment]
@eLBati 
```
You try to install module 'l10n_it_withholding_tax' that depends on module 'account_voucher_cash_basis'.
But the latter module is not available in your system.
```

**Comment by eLBati** [comment]
@lepistone this PR depends on https://github.com/OCA/account-payment/pull/18

**Comment by Giuliano69** [comment]
Hi elbati,
I see your PR  depends on PR 18 (account_vat_on_payment) to pass Travis test.
I see that  the PR18 has passed checks,
Are your PR and PR 18,  ready to merge the italian 10n_italy ?



Now that PR 73 (vat registries) and PR108 (vat statement) merged in PR 124 https://github.com/OCA/l10n-italy/pull/124
Are there any chance  your PR will change dependency on PR 108  too, to reduce modules intradependency ?

**Comment by eLBati** [comment]
As l10n_it_withholding_tax depends on 'account_voucher_cash_basis', if
you want to speed up this PR, you should make a review on the PR
related to 'account_voucher_cash_basis'

On 07/08/2015, Giuliano69 <notifications@github.com> wrote:
> Hi elbati,
> I see your PR  depends on PR 18 (account_vat_on_payment) to pass Travis
> test.
> I see that  the PR18 has passed checks,
> Are your PR and PR 18,  ready to merge the italian 10n_italy ?
>
>
>
> Now that PR 73 (vat registries) and PR108 (vat statement) merged in PR 124
> https://github.com/OCA/l10n-italy/pull/124
> Are there any chance  your PR will change dependency on PR 108  too, to
> reduce modules intradependency ?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/OCA/l10n-italy/pull/72#issuecomment-128528739


-- 
Lorenzo Battistini
Tel (CH): +41 91 210 23 40
Tel (IT): +39 011 198 25481
http://www.agilebg.com
https://github.com/eLBati

**Comment by sergiocorato** [comment]
@eLBati : anyway, if you watch [this] (https://github.com/sergiocorato/LibrERP-61/blob/master/l10n_it_withholding_tax/__openerp__.py) implementation, you can get rid of account_voucher_cash_basis dependency. For v.8, I'll push in a few weeks I think.

**Comment by eLBati** [comment]
Green :green_heart: 
Please review

File: l10n_it_withholding_tax/README.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
```
Comment (tafaRU) [inline l10n_it_withholding_tax/README.rst]: Equal signs are missing, please add them.

File: l10n_it_withholding_tax/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+{
+    'name': "Italian Localisation - Withholding tax",
+    'version': '1.0',
```
Comment (tafaRU) [inline l10n_it_withholding_tax/__openerp__.py]: Please update version according to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

File: l10n_it_withholding_tax/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "depends": ['account_voucher_cash_basis'],
+    "data": [
+        'account_view.xml', ],
+    "demo": [
+        'account_demo.xml',
+    ],
+    'test': [
+        'test/purchase_payment.yml',
+    ],
+    "active": False,
```
Comment (tafaRU) [inline l10n_it_withholding_tax/__openerp__.py]: This key is False by default so you could remove it.

Comment (lepistone) [inline l10n_it_withholding_tax/__openerp__.py]: Worse, it's misleading: `active` means `auto_install` IIRC. Please remove the line.

Comment (tafaRU) [inline l10n_it_withholding_tax/__openerp__.py]: Yes, indeed! What I wanted to say was that it's replaced by auto_install which is False by default :wink:

**Comment by eLBati** [comment]
@tafaRU thanks, done

**Comment by archetipo** [comment]
@eLBati please rebase

**Comment by eLBati** [comment]
@archetipo rebased. thanks

**Comment by eLBati** [comment]
La nuova versione del modulo si trova su https://github.com/OCA/l10n-italy/pull/194 (PR ancora da sistemare)
Se nessuno è contrario al passaggio al nuovo modulo, chiuderei questa PR

**Comment by dcorio** [comment]
@eLBati per come la vedo è un peccato perdere la tua versione del modulo. potrebbe essere utile rinominare questi in l10n_it_withholding_tax_minimal ad esempio.
La nuova versione è completa ma anche estremamente complessa e sovradimensionata per chi ha esigenze minime.

**Comment by eLBati** [comment]
Ok, allora chiudo questa in favore di https://github.com/OCA/l10n-italy/pull/198
(ho scritto nel README l'incompatibilità)

**Pull request closed without merging**